### PR TITLE
improve namespace and resource selector docs

### PR DESCRIPTION
**Pull request by lihongyan1**
fix description of `namespaceSelector` and `resourceSelector` to unify the empty and null

### CODE REVIEW

File: deploy/crds/common/monitoring.rhobs_monitoringstacks.yaml [hunk 1]
```diff
@@ -56,9 +56,10 @@ spec:
                 - error
                 type: string
               namespaceSelector:
-                description: Namespace selector for Monitoring Stack Resources. If
-                  left empty the Monitoring Stack will only match resources in the
-                  namespace it was created in.
+                description: 'Namespace selector for Monitoring Stack Resources. Set
```
Comment (lihongyan1) [inline deploy/crds/common/monitoring.rhobs_monitoringstacks.yaml:59]: @sthaha Not sure why the `single quote` is added after description

Comment (sthaha) [reply deploy/crds/common/monitoring.rhobs_monitoringstacks.yaml:59]: I don't think this matters as long as the string is valid. (and the test pass)

File: pkg/apis/monitoring/v1alpha1/types.go [hunk 2]
```diff
@@ -58,14 +58,15 @@ type MonitoringStackSpec struct {
 	LogLevel LogLevel `json:"logLevel,omitempty"`
 
 	// Label selector for Monitoring Stack Resources.
-	// Set to the empty LabelSelector ({}) to monitoring everything.
-	// Set to null to disable service discovery.
+	// Set to the empty label selector (resourceSelector: {}) to monitoring everything.
+	// Set to null (resourceSelector:) to disable service discovery.
```
Comment (sthaha) [inline pkg/apis/monitoring/v1alpha1/types.go:62]:
```suggestion
	// To monitor everything, set to a empty map selector. E.g. resourceSelector: {}
	// To disable service discovery, set to null. E.g. resourceSelector:
```

File: pkg/apis/monitoring/v1alpha1/types.go [hunk 3]
```diff
@@ -58,14 +58,15 @@ type MonitoringStackSpec struct {
...
+	// To monitor everything, set to an empty map selector. E.g. resourceSelector: {}
+	// To disable service discovery, set to null. E.g. resourceSelector:
 	// +optional
 	// +nullable
 	ResourceSelector *metav1.LabelSelector `json:"resourceSelector"`
 
 	// Namespace selector for Monitoring Stack Resources.
-	// If left empty the Monitoring Stack will only match resources in the namespace it was created in.
+	// Set to null (namespaceSelector:) the Monitoring Stack will only monitor resources in the namespace it was created in.
+	// Set to the empty namespace selector (namespaceSelector: {}) means monitor everything.
```
Comment (sthaha) [inline pkg/apis/monitoring/v1alpha1/types.go:69]: Could you please change this as well to match the style above? 
E.g. 
```
// To only monitor resources in the same namespace as the stack, set this to `null`. E.g. namespaceSelector: 
// To monitor resources in all namespaces, set the selector to empty map ({}). E.g. namespaceSelector: {}

**Review by sthaha — APPROVED** [review]

**Pull request merged**